about/release_policy.rst

```diff
@@ -158,3 +195,55 @@ The 3.6 release is still planned and should be the last stable branch of Godot
 as long as users still need it (due to missing features in Godot 4.x, or
 having published games which they need to keep updating for platform
 requirements).
+
+What are the criteria for compatibility across engine versions?
+---------------------------------------------------------------
+
+.. note::
+
+    This section is intended to be used by contributors to determine which
+    changes are safe for a given release. The list is not exhaustive; it only
+    outlines the most common situations encountered during Godot's development.
+
+The following changes are acceptable in patch releases:
+
+- Fixing a bug that has no major negative impact on most projects, such as a
+  visual or physics bug. Godot's physics engine is not deterministic, so physics
+  bug fixes are not considered to break compatibility. If fixing a bug has a
+  negative impact that could impact a lot of projects, it should be made
+  optional (e.g. using a project setting or separate method).
+- Adding a new optional parameter to a method.
+- Small-scale editor usability tweaks.
+
+The following changes are acceptable in minor releases, but not patch releases:
```

Comment:
I would add here bugfixes which significantly change the behavior or the output, with the aim to meet user expectations better. Compared to patch releases where we may favor keeping a buggy behavior so we don't break existing projects which likely rely on the bug or implement a workaround. (E.g. https://github.com/godotengine/godot/pull/78727)